homeassistant/components/dsmr/sensor.py

```diff
@@ -409,3 +444,105 @@ def unit_of_measurement(self):
         unit = self.get_dsmr_object_attr("unit")
         if unit:
             return f"{unit}/{TIME_HOURS}"
+
+
+class Timer:
+    """async Timer object with reset.
+
+    a resettable timer is not present in asyncio, however this integration
+    needs one to timeout faulty connections. It includes
+    garbage collection uses loop.call_later.
+
+    """
+
+    def __init__(self, delay, timer_callback, *args, loop=None):
+        """Create a timer with confingurable timeout and callback.
+
+        Has no timerhandle  on creation(will be created on start) and the
+        callback and delay will be used to create this timerhandle.
+
+        """
+        self.timerhandle = None
+        self.set_timeout(delay)
+        self.set_callback(timer_callback)
+        self.args = args
+        if loop:
+            self.loop = loop
+        else:
+            self.loop = asyncio.get_event_loop()
```

Comment:
Let's drop support for passing in `args` and `loop`. Use `asyncio.get_running_loop()` to get the loop.